### DISCUSSION TITLE
weird filtering behaviour

### DISCUSSION THREAD

**Discussion by eggehad** [FoobarOy/foomuuri#57] [General]
I absolutely **LOVE** this tool.  Just discovered it about 3 days ago, and I've been playing with setting up different firewall configurations. One little experiment I did, was to take your "bidirectional router" example and modify it with my interfaces, then run a port scan (grc.com) against the firewall created, and got the port scan results in "Screenshot of Bidirectional filter with single ISP."  Then later on, I did the same using the multi-ISP version of the same  "Screenshot of Bidirectional filter with dual ISP".  Curiously, the single iSP version shows all the ports as stealth, but the dual ISP version shows a shifting pattern of closed and stealth ports,  changing every time I run it.  Does anyone understand whats happening?  Why do all the ports show as stealth with the single ISP, and as a mixture of stealth and closed for the dual ISP version?  Is there a way to make the dual ISP version also show stealth mode for all the scanned ports?
Thanks!
-david

![Screenshot of Bidirectional filter with dual ISP](https://github.com/user-attachments/assets/fc950b8f-11cc-4aec-bccc-07bee9bc58f9)
![Screenshot of Bidirectional filter with single ISP](https://github.com/user-attachments/assets/1675d425-d234-4f5a-9bc4-a0006c2def99)

**Comment by kimheino** [comment]
Hi,

I tested grc.com's scanner for my site and it was mostly green / stealth, just like you got.

What do you mean by "multi-ISP version"? Do you have multi-ISP setup, two uplinks? I guess uplink #1 is configured correctly but uplink #2 is not. Do you have same IP address for both? In Foomuuri you should include both interfaces to public zone:

```
zone {
  public   eth0 eth1
}
```

Maybe share your foomuuri.conf and tell more about your network setup?
